### DISCUSSION TITLE
2D Networks transfer learning

### DISCUSSION THREAD

**Discussion by LSnyd** [Project-MONAI/tutorials#676] [Q&A]
Hi all, 
is there a way to add pretrained weights of imagenet for instance to the Monai 2D UNet? 
Thank you!

**Comment by Nic-Ma** [comment]
Hi @LSnyd ,

Thanks for your interest here.
I think we have pretrained weights option for many networks, for example:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/networks/nets/efficientnet.py#L710
@yiheng-wang-nv can help share more details about your target network.

Thanks.

**Reply by LSnyd** [reply]
Thank you. I am using the Monai UNet implementation and it doesn't seem like there is an option for transfer learning. 
@yiheng-wang-nv Do you know if there is a way to add this to my implementation? I am not sure if the manual adding works, cause the imageNet UNet may use different keys to store the weights/network than the UNet implementation?

**Reply by yiheng-wang-nv** [reply]
Hi @LSnyd , if the imageNet UNet you mentioned has same layers as MONAI's UNet, you may need to do some mapping manipulations to convert the weights. There is an example of how to do that for senet:
https://github.com/Project-MONAI/MONAI/blob/e093d0b1572f8b5b49265f856955cb63fcd39bad/monai/networks/nets/senet.py#L284
